src/hooks/use-sentinels-historical-data.js

```diff
@@ -1,25 +1,66 @@
 import axios from 'axios'
 import { useEffect, useState } from 'react'
+import { groupBy } from 'lodash'
+
+const calculateFee = ({ epochs, data, keyAppender = '' }, _isValid) => {
+  let sum = []
+  for (const epoch of epochs) {
+    const element = data[`${epoch}${keyAppender}`]
+
+    if (!_isValid(element)) {
+      sum.push(0)
+      continue
+    }
+
+    sum.push(
+      Object.values(element.fees).reduce((_acc, { amount_usd }) => {
+        _acc += amount_usd
+        return _acc
+      }, 0)
+    )
+  }
+  return sum
+}
 
 const useSentinelsHistoricalData = () => {
   const [numberOfNodes, setNumberOfNodes] = useState([])
   const [epochs, setEpochs] = useState([])
-  const [accruedFees, setAccruedFees] = useState([])
+  const [accruedNodesFees, setAccruedNodesFees] = useState([])
+
+  const handlePartialData = (data) =>
+    typeof data === 'object'
+      ? data
+      : typeof data === 'string'
+      ? JSON.parse(data.replace(/: ,/g, ': "NaN",').replace(/: }/g, ': "NaN"}'))
```

Comment:
Put a comment explaining why this processing is needed as it's not immediate. Would you remember it in a year?